src/app/action-items/action-items.component.ts

```diff
@@ -57,7 +62,49 @@ export class ActionItemsComponent implements OnInit {
       .subscribe(item => {
         if (!!item) {
           this.retrieveActionItems();
+          this.justAddedItemId = this.dataSource[this.dataSource.length - 1].id;
+          this.router.navigate(['/items'], { queryParams: { id: this.justAddedItemId } });
+          setTimeout(() => {
+            item = this.elRef.nativeElement.querySelector(`[data-id="${this.justAddedItemId}"]`);
+            this.triggerScrollTo();
+          }, 1000);
```

Comment:
FIX: This will work because of the assumption that refreshing the action item list will take 1 second. If you go to the `tasks.service.ts` and increase the timeout value, for example:
```
  getAllItems(): Observable<ActionItem[]> {
    return new Observable(observer => {
      setTimeout(() => {
        observer.next(this.dataTable);
      }, 3000); <--------------- 3 seconds
    });
```

You will notice that scroll functionality won't work. In real-world scenario we never know how long we need to wait for the API response, so we cannot do assumption like that here.